### ISSUE TITLE
SSL Hostname validation seems to incorrectly fail on certain certs with wildcard SANs

### ISSUE THREAD

**Issue by jeffmlevine** [akka/akka-http#4431]
I have a 3rd party API I'm trying to access with akka-http (10.5.3) and it is failing with a CertificateException

`No subject alternative DNS name matching em.engageware.api.onereach.ai found.`

This is because the code believes that the list of SANs is 

`*.execute-api.us-east-1.amazonaws.com`

while the hostname is: 

`em.engageware.api.onereach.ai`

Normally, I'd think this is a cert problem and I'd go to the vendor to see if they can change the cert.  However, accessing this endpoint via Chrome, curl, Postman, and Java (8 or 17 via RestTemplate) all work fine without any SSL errors.   I also checked online SSL checkers, and they report no errors.  Additionally, the SANs look correct there:
```
Common name: *.engageware.api.onereach.ai
SANs: *.engageware.api.onereach.ai, *.svc.engageware.api.onereach.ai, *.onereach.ai, *.hitl.engageware.api.onereach.ai, *.fileshooks.engageware.api.onereach.ai, *.auth.engageware.api.onereach.ai, *.em.engageware.api.onereach.ai, *.store.engageware.api.onereach.ai
```

Also, as a workaround, disabling hostname verification does get around the issue, but I'd like to figure out why it's failing and not disable this security check.

Simple code to reproduce:
```
    try { 
      val req = HttpRequest(
        method = HttpMethods.GET,
        uri = "https://em.engageware.api.onereach.ai/http/12abf4bf-123b-48bc-8dc2-084f18feebcb/test")
      val respF = Http().singleRequest(req)
    } catch {
      case e: Exception =>
        e.printStackTrace()
        println(e.getMessage)
    }

```
That test URL should be active and should produce a Hello World response.

I tested under both akka-http 10.5.3  and 10.7.3 running Java 17 and both have the issue.

**Comment by johanandren** [comment]
I can't reproduce this, I tried JDK Temurin-11.0.21+9, Temurin-17.0.8.1+1, Temurin-21.0.2+13 and I get a OK response from all three. I wonder if it could be related to something broken in earlier patch versions of the Java releases?

There is no Akka HTTP 10.7.3 yet, so I expect you meant 10.6.3.

For future reference, reproducer did not wait for response, here's a full reproducer:
```scala
import akka.actor.ActorSystem
import akka.http.scaladsl.Http
import akka.http.scaladsl.model.HttpMethods
import akka.http.scaladsl.model.HttpRequest

import scala.concurrent.Await
import scala.concurrent.duration.DurationInt

object Test {

  def main(args: Array[String]): Unit = {
    implicit val system = ActorSystem("test")
    try {
      val req = HttpRequest(
        method = HttpMethods.GET,
        uri = "https://em.engageware.api.onereach.ai/http/12abf4bf-123b-48bc-8dc2-084f18feebcb/test")
      val respF = Http().singleRequest(req)
      val res = Await.result(respF, 10.seconds)
      println("response " + res.status)
    } catch {
      case e: Exception =>
        e.printStackTrace()
        println(e.getMessage)
    }
  }

}
```

**Comment by jeffmlevine** [comment]
@johanandren I really appreciate the quick response!  

First, you are correct, that was my typo (10.6.3, not 10.7.3), and I did accidentally delete the Await line when I was editing my response down to the minimum.

As for the issue, I started over from a new project and lo and behold everything worked! 
Given that knowledge, I started hunting down in some libraries we have and some legacy code was setting on startup:
` System.setProperty("jsse.enableSNIExtension", "false");`
apparently due to some misbehaving 3rd party site around the 2015 timeframe which has long since been replaced.

Once I removed that line, everything was working fine.   That was definitely the last thing I expected to discover, but I'm definitely grateful it wasn't some obscure bug in the akka-http library.

Again, I really appreciate your help here and sorry for the false alarm!
